src/UsgsAstroLsSensorModel.cpp

```diff
@@ -1,2746 +1,2574 @@
-    const csm::EcefCoord& ground_pt, double desired_precision,
-    double* achieved_precision, csm::WarningList* warnings) const {
-  MESSAGE_LOG(
-      "Computing groundToImage(No adjustments) for {}, {}, {}, with desired "
-      "precision {}",
-      ground_pt.x, ground_pt.y, ground_pt.z, desired_precision);
-
-  // The public interface invokes the private interface with no adjustments.
-  return groundToImage(ground_pt, _no_adjustment, desired_precision,
-                       achieved_precision, warnings);
-}
-
-//***************************************************************************
-// UsgsAstroLsSensorModel::groundToImage (internal version)
-//***************************************************************************
-csm::ImageCoord UsgsAstroLsSensorModel::groundToImage(
-    const csm::EcefCoord& groundPt, const std::vector<double>& adj,
-    double desiredPrecision, double* achievedPrecision,
-    csm::WarningList* warnings) const {
-  // Search for the line, sample coordinate that viewed a given ground point.
-  // This method first uses a linear approximation to get an initial point.
-  // Then the detector offset for the line is continuously computed and
-  // applied to the line until we achieve the desired precision.
-
-  csm::ImageCoord approxPt;
-  computeLinearApproximation(groundPt, approxPt);
-
-  std::vector<double> detectorView;
-  double detectorLine = m_nLines;
-  double detectorSample = 0;
-  double count = 0;
-  double timei;
-
-  while (abs(detectorLine) > desiredPrecision && ++count < 15) {
-    timei = getImageTime(approxPt);
-    detectorView = computeDetectorView(timei, groundPt, adj);
-
-    // Convert to detector line
-    detectorLine = m_iTransL[0] + m_iTransL[1] * detectorView[0] +
-                   m_iTransL[2] * detectorView[1] + m_detectorLineOrigin -
-                   m_startingDetectorLine;
-    detectorLine /= m_detectorLineSumming;
-
-    // Convert to image line
-    approxPt.line += detectorLine;
-  }
-
-  timei = getImageTime(approxPt);
-  detectorView = computeDetectorView(timei, groundPt, adj);
-
-  // Invert distortion
-  double distortedFocalX, distortedFocalY;
-  applyDistortion(detectorView[0], detectorView[1], distortedFocalX,
-                  distortedFocalY, m_opticalDistCoeffs, m_distortionType,
-                  desiredPrecision);
-
-  // Convert to detector line and sample
-  detectorLine = m_iTransL[0] + m_iTransL[1] * distortedFocalX +
-                 m_iTransL[2] * distortedFocalY;
-  detectorSample = m_iTransS[0] + m_iTransS[1] * distortedFocalX +
-                   m_iTransS[2] * distortedFocalY;
-  // Convert to image sample line
-  double finalUpdate =
-      (detectorLine + m_detectorLineOrigin - m_startingDetectorLine) /
-      m_detectorLineSumming;
-  approxPt.line += finalUpdate;
-  approxPt.samp =
-      (detectorSample + m_detectorSampleOrigin - m_startingDetectorSample) /
-      m_detectorSampleSumming;
-
-  double precision =

```

Comment:
I removed this because it wasn't used, but I find the lack of any logic in here determining whether achievedPrecision is true or not confusing. Any ideas?

Reply:
The next block reports the final achieved precision.